bld/namelist_files/namelist_definition_clm4_5.xml

```diff
@@ -1071,6 +1071,12 @@ Toggle for mexico city specific logic.
 Max number of plant functional types in naturally vegetated landunit.
 </entry>
 
+<entry id="n_dom_soil_patches" type="integer" category="physics"
+       group="clm_inparm"
+       valid_values="0,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,78" value="78">
+Number of dominant soil patches; determines number of active soil patches.
+</entry>
```

Comment:
I did not need to limit valid_values to 16. The code should handle any value > 0.  Is there a way to set valid_values="non-negative" somehow?

On the other hand, I think that this capability loses it's practical purpose for n_dom_soil_patches > 16 because you're unlikely to save significant resources at that point. Also at that point users might as well use the collapse_crop_types capability and modify merge2clmpft as they need.

Reply:
There isn't a way to say non-negative right now. But, I think being explicit like this is probably good. Although you should remove the option to run with 0!

Reply:
Done.

Reply:
I brought back the option to run with 0 and removed the option to run with 78. This way standard testing can PASS when the user has not set n_dom_soil_patches in their namelist.